src/util/key.rs

```diff
@@ -128,6 +128,7 @@ impl PrivateKey {
         let mut ret = [0; 34];
         ret[0] = match self.network {
             Network::Bitcoin => 128,
+            Network::Signet => 217,
```

Comment:
Verifiable here: https://github.com/kallewoof/bitcoin/blob/2b9db9ac0936cf278ad27c59497d9feec61f6440/src/chainparams.cpp#L336

Reply:
You forgot to also add the parsing logic for this in the `from_wif` method.